### ISSUE TITLE
edge case: duplication of schema warnings

### ISSUE THREAD

**Issue by quadrophobiac** [theodi/csvlint.rb#148]
when working with the gem, if an empty JSON file is passed as a schema then the :extra_column warning will be reported multiple times because schema.fields will always be less than any CSV row passed to the schema for evaluation

**Comment by quadrophobiac** [comment]
Digging deeper I have realised that this warnings problem is a symptom of a problem which was identified in CSVlint.io bug reports regarding empty and malformed JSON schemas

Ruby JSON.parse doesn't raise an error if it returns an empty JSON object

The Schema class could be rewritten thusly:
```ruby
    def Schema.from_json_table(uri, json)
      fields = []
      json["fields"].each do |field_desc|
        fields << Csvlint::Field.new( field_desc["name"] , field_desc["constraints"],
          field_desc["title"], field_desc["description"] )
      end if json["fields"]
      return Schema.new( uri , fields, json["title"], json["description"] )
    end

    def Schema.load_from_json_table(uri)
      begin
        json = JSON.parse( open(uri).read ).empty? ? nil : JSON.parse( open(uri).read )
        return Schema.from_json_table(uri,json)
      rescue
        return Schema.new(nil, [], "malformed", "malformed")
      end
    end
```
which will ensure that an error is raised when `from_json_table` each fails (due to receiving nil for fields)

However any fix like this could affect the rails app - currently the duplication of warnings doesn't surface in the rails app as we catch Schema errors with a "Empty Schema" error and a "Malformed Schema" else if
app/models/validation.rb:121 & 122 - empty,  
app/models/validation.rb:114 - 118 - malformed
